streaming/format/mds/shard.py

```diff
@@ -133,16 +185,18 @@ def decode_sample(self, data: bytes) -> Dict[str, Any]:
         """
         sizes = []
         idx = 0
-        for key, size in zip(self.column_names, self.column_sizes):
-            if size:
-                sizes.append(size)
+        for column in self.columns:
```

Comment:
We could, but not as elegantly as doing it two passes, because the format is like `[sizes of dynamically sized fields only] [all fields]`, and I really doubt it would make a perf difference? But worth benchmarking in future work.